fdbserver/TLogServer.actor.cpp

```diff
@@ -1817,26 +1817,23 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 	// if tLog locked for recovery, return an empty message at the cluster recovery version
 	// if requested version is greater than any received.
 	state Optional<Version> clusterRecoveryVersion = Optional<Version>();
-	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR && logData->stopped()) {
-		if (!self->clusterRecoveryVersion.get().present()) {
-			TraceEvent("TLogPeekMessages0", self->dbgid)
-				.detail("LogId", logData->logId)
-				.detail("ReqReturnIfBlocked", reqReturnIfBlocked ? 1 : 0)
-				.detail("ReqBegin", reqBegin)
-				.detail("Version", logData->version.get())
-				.detail("RecoveredAt", logData->recoveredAt);
-			if (logData->version.get() < reqBegin) {
-				TraceEvent("NoClusterRecoveryVersion",self->dbgid).detail("X",reqReturnIfBlocked);
-				wait(self->clusterRecoveryVersion.onChange());
-				// wait(timeout(self->clusterRecoveryVersion.onChange(),10,Void()));
-				// ASSERT(self->clusterRecoveryVersion.get().present());
-				TraceEvent("YesClusterRecoveryVersion",self->dbgid);
-			}
+	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST && logData->stopped()) {
+		DebugLogTraceEvent("TLogPeekMessagesWaitForCrv", self->dbgid)
+		    .detail("Tag", reqTag.toString())
+		    .detail("LogId", logData->logId)
+		    .detail("ReqReturnIfBlocked", reqReturnIfBlocked ? 1 : 0)
+		    .detail("ReqBegin", reqBegin)
+		    .detail("Version", logData->version.get())
+		    .detail("RecoveredAt", logData->recoveredAt);
+
+		if (!logData->clusterRecoveryVersion.get().present()) {
+			wait(logData->clusterRecoveryVersion.onChange());
 		}
-	}
-	if (logData->stopped() && logData->version.get() < reqBegin && self->clusterRecoveryVersion.get().present()) {
-		clusterRecoveryVersion = self->clusterRecoveryVersion.get().get();
-		TraceEvent("TLogPeekMessagesClusterRecoveryVersion", self->dbgid).detail("Version", clusterRecoveryVersion.get());
+		clusterRecoveryVersion = logData->clusterRecoveryVersion.get().get() + 1;
+		TraceEvent("TLogPeekMessagesClusterRecoveryVersion", self->dbgid)
+		    .detail("LogId", logData->logId)
+		    .detail("LogEpoch", logData->epoch())
+		    .detail("CRV", clusterRecoveryVersion.get());
```

Comment:
Here is a reader of the recovery version: this is not correct (as the async variable has the bumped up version), right?